### ISSUE TITLE
sanoid WIP

### ISSUE THREAD

**Issue by deliciouslytyped** [NixOS/nixpkgs#71423]
```nix
{lib, stdenv, fetchFromGitHub, bash, makeWrapper, pv, mbuffer, perl, zfs, fetchpatch}:
let
  perlp = perl.withPackages (p: [ p.CaptureTiny p.ConfigIniFiles ]);
  binaryDeps = [ pv mbuffer perlp zfs ]; #TODO assert version is inline with kernel?
  fetchGitHubJSON = file: fetchFromGitHub { inherit (lib.importJSON file) owner repo rev sha256; };
in
  stdenv.mkDerivation {
    pname = "sanoid";
    #version = "v2.0.2";
    version = "master";

    # example: nix-shell -p nix-prefetch-github --run 'nix-prefetch-github jimsalterjrs sanoid --rev master > sanoid.json' 
    src = fetchGitHubJSON ./sanoid.json; 

    nativeBuildInputs = [ makeWrapper ];

    patches = [
      (fetchpatch {
        name = "fix-wait-any.patch";
        url = "https://patch-diff.githubusercontent.com/raw/jimsalterjrs/sanoid/pull/399.patch";
        sha256 = "sha256:16hlwcbcb8h3ar1ywd2bzr3h3whgbcfk6walmp8z6j74wbx81aav";
        })
      ];

    #TODO wrapphase
    installPhase = ''
      runHook preInstall
      cp -r . $out

      sed -ie "s,.\+bin/perl,#!${perlp}/bin/perl," $out/syncoid # stolen from x2goserver
      wrapProgram $out/syncoid \
        `# hmm this might cause problems for user configs` \
        --add-flags "--configdir=\"$out\"" \
        --prefix PATH : ${lib.makeBinPath binaryDeps}

      sed -ie "s,.\+bin/perl,#!${perlp}/bin/perl," $out/sanoid # stolen from x2goserver
      wrapProgram $out/sanoid \
        `# hmm this might cause problems for user configs` \
        --add-flags "--configdir=\"$out\"" \
        --prefix PATH : ${lib.makeBinPath binaryDeps}

      sed -ie "s,.\+bin/perl,#!${perlp}/bin/perl," $out/findoid # stolen from x2goserver
      wrapProgram $out/findoid \
        --prefix PATH : ${lib.makeBinPath binaryDeps}


      runHook postInstall
      '';

    }
```

cc @Nekroze

**Comment by lopsided98** [comment]
I have my own version of this package I have been using for a couple of years: https://github.com/lopsided98/nixos-config/blob/master/pkgs/sanoid/default.nix

I also have NixOS modules for [sanoid](https://github.com/lopsided98/nixos-config/blob/master/modules/services/backup/sanoid.nix) and [syncoid](https://github.com/lopsided98/nixos-config/blob/master/modules/services/backup/syncoid.nix).

**Comment by lopsided98** [comment]
See #72060

**Comment by lopsided98** [comment]
This can be closed now that #72060 is merged.

**Comment by deliciouslytyped** [comment]
Awesome!
